src/ploomber_core/telemetry/telemetry.py

```diff
@@ -682,7 +684,19 @@ def wrapper(*args, **kwargs):
 
                 try:
                     if payload:
-                        result = func(_payload, *args, **kwargs)
+                        if "self" in inspect.signature(func).parameters:
```

Comment:
@edublancas 
I am not 100% confident that I am using signature correct, please check this line~

Reply:
we should check the position as well: `"self" == list(s.parameters)[0]`. Let's create this in a separate method so we implement test cases.

also, let's move this check a few lines above, if we do it here, it'll always run whenever we call the method, but we only need this check once, so if we move it right below `def _log_call`, it'll only be performed once at import time:


```python
 def _log_call(func):
    is_method = first_arg_is_self(func)
    
```

Reply:
Fix in [890f4bf](https://github.com/ploomber/core/pull/45/commits/890f4bf6b5e563c950cc3cd53fb49ebf37bb2f09)